### PR TITLE
emit TRANSFER on steps, debug fixed-quote-policy

### PR DESCRIPTION
**Pull request by ggobugi27**
- Add p-protocol in `fungible-utils`
- Emit `TRANSFER` on pact steps
- remove `timeout` at offer
- fixed-quote-policy (in progress)

### CODE REVIEW

**Comment by ggobugi27** [comment]
@sirlensalot  
https://github.com/kadena-io/hft/blob/4ac45098081d0ba79ada3620a27f59f67e8ad640/pact/fixed-quote-policy.pact#L114
need to be placed in a different place, since it's executed inside a defcap `BUY`.

File: pact/hft.pact [hunk 1]
```diff
@@ -452,7 +451,7 @@
     )
     (step-with-rollback
       (with-capability (SALE token seller amount timeout (pact-id))
-        (offer token seller amount timeout))
+        (offer token seller amount))
```
Comment (sirlensalot) [inline pact/hft.pact:454]: btw let's change `token` to `id` for all this stuff

Comment (sirlensalot) [reply pact/hft.pact:454]: also see response to your comment, instead of `sale` for caps let's use `sale-id` to not conflict with the name of the defpact

**Review by sirlensalot — COMMENTED** [review]
Regarding your question, we need to move `policy::enforce-sale` call out of `BUY` and into `buy`.

File: pact/hft.repl [hunk 2]
```diff
@@ -352,3 +353,325 @@
...
+    (hft.hft.OFFER "project-0" "bob" 0.1 10)
+    (hft.hft.BUY "project-0" "bob" "buyer" 0.1 110 (hash "sale-tx1"))
+    ]}
+  ])
+
+(env-data
+ { 'buyer: "buyer",
+   'buyer-guard: {"keys": ["buyer"], "pred": "="}
+ })
+
```
Comment (sirlensalot) [inline pact/hft.repl:379]: We need a failure case for init-sale failing because sale-guard not signed

File: pact/hft.repl [hunk 3]
```diff
@@ -352,3 +353,325 @@
...
+(expect "Seller is debited, sale account is credited"
+  [2.9 0.1]
+  (map (get-balance 'project-0) ['bob (sale-account)]))
+
+(expect "OFFER events"
+  [{"name": "hft.hft.OFFER","params": ["project-0" "bob" 0.1 110]}
+   {"name": "hft.hft.SALE","params": ["project-0" "bob" 0.1 110 (hash "sale-tx1")]}
+   {"name": "hft.hft.TRANSFER","params": ["project-0" "bob" (sale-account) 0.1]}]
+  (map (remove 'module-hash) (env-events true)))
+
```
Comment (sirlensalot) [inline pact/hft.repl:397]: We need a failure case for `enforce-sale` failing because sale-guard not signed

**Pull request merged**